jyulb/internal/common/tests/test_proxy_lattice.py

```diff
@@ -0,0 +1,133 @@
+"""Testing module for jyu-lb proxy-lattice data class."""
+import unittest
+import numpy as np
+import numpy.testing as np_test
+from simphony.core.cuba import CUBA
+from simphony.cuds.lattice import LatticeNode
+from simphony.cuds.abstractlattice import ABCLattice
+from simphony.core.data_container import DataContainer
+from jyulb.internal.common.proxy_lattice import ProxyLattice
+from jyulb.internal.common.domain import PyLattice, PyGeometry
+from jyulb.internal.common.domain import PyIsothermalData
+
+
+class ProxyLatticeTestCase(unittest.TestCase):
+
+    """Test case for ProxyLattice class."""
+
+    def setUp(self):
+        self.h = 0.1
+        self.nx = 10
+        self.ny = 20
+        self.nz = 30
+        self.name = 'Test'
+        self.type = 'Cubic'
+        self.bvec = (self.h, self.h, self.h)
+        self.size = np.array((self.nx, self.ny, self.nz), dtype=np.uint32)
+        self.orig = np.array((0, 0, 0), dtype=np.float64)
+
+        self.hl_data = DataContainer()
+        self.hl_data[CUBA.STATUS] = 0
+
+        self.lat = PyLattice(self.size, self.orig)
+        self.geom = PyGeometry(self.lat)
+
+        for index in np.ndindex(tuple(self.size)):
+            if index[0] > 0 and index[0] < self.nx-1:
+                ijk = np.array(index, dtype=np.uint32)
+                self.geom.set_material_ijk(ijk, ProxyLattice.FLUID_ENUM)
+
+        ivel = np.array((0.0, 0.0, 0.0), dtype=np.float64)
+        ifrc = np.array((0.0, 0.0, 0.0), dtype=np.float64)
+        self.fdata = PyIsothermalData(self.lat, 0.0, ivel, ifrc)
+
+    def test_construct_lattice(self):
+        """Construction of a lattice."""
+        proxy = ProxyLattice(self.name, self.type,
+                             self.bvec, self.geom, self.fdata)
+
+        proxy.data = self.hl_data
+
+        self.assertIsInstance(proxy, ABCLattice, "Error: not a ABCLattice!")
+
+        self.assertEqual(proxy.name, self.name)
+        self.assertEqual(proxy.type, self.type)
+        self.assertEqual(proxy._geom, self.geom)
+        self.assertEqual(proxy._fdata, self.fdata)
+        self.assertEqual(proxy.data, self.hl_data)
+        np_test.assert_array_equal(proxy.size, self.size)
+        np_test.assert_array_equal(proxy.origin, self.orig)
+        np_test.assert_array_equal(proxy.base_vect, self.bvec)
+
+        try:
+            proxy.get_node((-1, 0, 0))
+        except IndexError:
+            pass
+        else:
+            raise AssertionError('Negative indices incorrectly accepted.')
+
+        test_node = LatticeNode((0, -1, 0))
+        try:
```

Comment:
Please use the specific assert methods provided by unittests https://docs.python.org/2/library/unittest.html#assert-methods. In this case I think: 
```python
with self.assertRaises(IndexError):
      proxy.update_node(test_node)
```